src/javascript/app_2/Utils/Validator/__tests__/declarative_validation_rules.spec.js

```diff
@@ -0,0 +1,319 @@
+                });
+
+                it('expects true if the value is not match', () => {
+                    expect(rules['not_equal'].func('txt_wrong', options)).to.be.true;
+                });
+            });
+        });
+
+        describe('.validMin', () => {
+            it('should return true if options.min is null, undefined, or 0', () => {
+                options.min = 0;
+                expect(rules['min'].func('', options)).to.be.true;
+            });
+            it('should return true if value.length >= options.min', () => {
+                options.min = 2;
+                expect(rules['min'].func('00', options)).to.be.true;
+            });
+            it('should return false if value.length < options.min', () => {
+                options.min = 2;
+                expect(rules['min'].func('0', options)).to.be.false;
+            });
+        });
+
+        describe('.validLength', () => {
+            it('should return true if value.length is in boundary', () => {
+                options.min = 1;
+                expect(rules['length'].func('00', options)).to.be.true;
+            });
+            it('should return false if value is smaller than minimum', () => {
+                options.min = 1;
+                expect(rules['length'].func('', options)).to.be.false;
+            });
+            it('should return false if value is bigger than maximum', () => {
+                options.max = 1;
+                expect(rules['length'].func('00', options)).to.be.false;
+            });
+            it('should return true if min and max is not set', () => {
+                options.max = 0;
+                expect(rules['length'].func('0', options)).to.be.true;
+            });
+        });
+
+        describe('.validNumber', () => {
+            // New lines seperates tests for different if statements.
+            it('should return true if value is empty', () => {
+                expect(rules['number'].func('', options)).to.be.true;
+            });
+
+            it('should return true if float', () => {
+                expect(rules['number'].func(0.0, options)).to.be.true;
+            });
+            it('should return true if integer', () => {
+                options['type'] = 'int';
+                expect(rules['number'].func(0, options)).to.be.true;
+            });
+            it('should return false if value is characters', () => {
+                expect(rules['number'].func('a', options)).to.be.false;
+            });
+            it('should return false if value is NaN', () => {
+                expect(rules['number'].func(NaN, options)).to.be.false;
+            });
+
+            it('should return false if value is float which does not exceed options.decimal', () => {
+                expect(rules['number'].func(0.002, options)).to.be.false;
+            });
+
+            it('should return false if min and max equals to one another while value is not equal to min', () => {
```

Comment:
isnt this the same as out of boundary ?

Reply:
yea, however, the original statement checks for that.